otaclient/ota_proxy/orm.py

```diff
@@ -90,29 +100,40 @@ def __get__(self, obj, objtype: type) -> FV:
         """Descriptor accessed via bound instance."""
         ...
 
-    def __get__(self, obj, objtype=None) -> Union[FV, "ColumnDescriptor[FV]"]:
+    def __get__(self, obj, objtype=None) -> Union[FV, Self]:
+        # try accessing bound instance attribute
         if not self._skipped and obj is not None:
             if isinstance(obj, type):
-                return self  # bound inst is type, treated same as accessed via class
+                return self  # bound inst is type(class access mode), return descriptor itself
             return getattr(obj, self._private_name)  # access via instance
-        return self  # access via class, return the descriptor
+        return self  # return descriptor instance by default
 
     def __set__(self, obj, value: Any) -> None:
         if self._skipped:
-            return
-        # handle dataclass's default value setting behavior and NULL type assignment
-        if isinstance(value, type(self)) or self.field_type == NULL_TYPE:
+            return  # ignore value setting on skipped field
```

Comment:
Is this __set__ function compatible with previous version, especially `value is None`?
Or any logical fix is introduced?

Reply:
The logic is the same, just the implementation is simplified. If the input `value` is None, or field type is `NULL_TYPE`, or dataclass initializing, the default value will be set.

This method is refined and simplify as `NULL_TYPE` is no-longer type alias of python's built-in `None` type. Previously in runtime `type(None)` is the same as `NULL_TYPE`, resulting we have to implement special mechanism to distinguish them when we have input value as `None`. 
This is not required anymore with this PR as `NULL_TYPE` becomes real type that different from `None` type.

Previously:
`type(None) == NULL_TYPE`, so we cannot use judgement like `self.field_type is NULL_TYPE`,
With PR:
`type(None) != NULL_TYPE`, `self.field_type is NULL_TYPE` can be used now.